### PR TITLE
add functionality use_mocks_only

### PR DESCRIPTION
**Pull request by risentveber**
Add use mocks only functionality. When it is enabled and there is no such route it throws error.

### CODE REVIEW

**Comment by A** [comment]
LGTM 👍  @tbranyen could you take a look on it?

**Comment by maxkoryukov** [comment]
@tbranyen , @A, it is really useful stuff, review plz

File: test.js [hunk 1]
```diff
@@ -125,6 +125,20 @@ describe('superagent mock', function() {
...
+
+        should.throws(function() {
+            request
+                .get('http://example.com')
+                .query({ foo: 'bar' })
+                .end(function(err, res) {});
+        }, /Unmocked url/);
+        done();
+
+        mock.disableUseMocksOnly();
```
Comment (tbranyen) [inline test.js:139]: Shouldn't the clean up code happen before we call `done()`? My assumption was that once `done()` was called, the next test starts running.

**Comment by tbranyen** [comment]
Looks good to me, just left a comment about the clean up code.

**Comment by maxkoryukov** [comment]
So we are just waiting for a fix from @risentveber ??

**Comment by tbranyen** [comment]
Looks like it, I'm a bit hesitant to merge in something that could break tests down the line.

**Pull request closed without merging**